src/apps/frontend/contexts/async.hook.ts

```diff
@@ -0,0 +1,50 @@
+import { useCallback, useState } from 'react';
+
+import {
+  UseAsyncResponse,
+  AsyncResult,
+  AsyncError,
+} from '../types';
+import { DefaultAsyncError } from '../types/async-operation';
+
+const useAsync = <T>(
+  asyncFn: (...args: any[]) => Promise<AsyncResult<T>>,
+): UseAsyncResponse<T> => {
+  const [result, setResult] = useState<T | undefined>(undefined);
+  const [isLoading, setLoading] = useState(false);
+  const [error, setError] = useState<AsyncError | undefined>(undefined);
+
+  const asyncCallback = useCallback(
+    async (...args) => {
+      setError(undefined);
+      setLoading(true);
+      try {
+        const response = await asyncFn(...args);
+
+        if (response.data) {
+          setResult({ ...response.data });
+        }
+
+        return response?.data;
+      } catch (e) {
+        const err = e?.response?.data?.httpStatusCode
+          ? new AsyncError(e?.response?.data)
+          : new DefaultAsyncError(e);
```

Comment:
async hook should not know anything about httpStatusCode

Reply:
Removed